tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -643,6 +643,17 @@ public function testInvalidInExpressionWithSingleValuedAssociationPathExpression
         );
     }
 
+    public function testCustomFunctionWithinInExpression() : void
+    {
+        $this->entityManager->getConfiguration()->addCustomStringFunction('FOO', MyAbsFunction::class);
+        $this->assertSqlGeneration(
+            "SELECT u FROM Doctrine\Tests\Models\Forum\ForumUser u WHERE u.username IN (FOO('Lo'), 'Lo', :name)",
```

Comment:
Why are you using strings in this example? It does not make a lot of sense to use an arithmetic function on a string, does it? Can you maybe write something closer to the real-world use-case you have?

Reply:
You can disregard my comment if that's too hard: I just read the EBNF for https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/dql-doctrine-query-language.html#arithmetic-expressions and now I understand it's not just about `2+2` :slightly_smiling_face:

Reply:
In my use case, the scenario looks like this: I have a database platform specific function that I want to use like `NVL2` from Oracle. For this I have created a user defined function for DQL in Doctrine and now I want to use this also for IN where conditions.

Reply:
OK. To me it's kind of weird that `ArithmeticPrimary` contains that many things, for instance `FunctionsReturningStrings`. I don't think this has much to do with arithmetic, but anyway, that's not really your issue.

Reply:
You could also add an arithmetic expression, it should not work before to do `IN (1+1)`